### ISSUE TITLE
Publish to Maven Central Repository

### ISSUE THREAD

**Issue by ceefour** [JorenSix/TarsosDSP#34]
It would be great if TarsosDSP can be published to Maven Central Repository, this will help a lot to make it easier to add TarsosDSP in a project.

Sonatype provides free hosting : http://central.sonatype.org/pages/ossrh-guide.html

**Comment by JorenSix** [comment]
Hi,

I have tried this (see the commit history) but this introduced a lot of extra cruft. Since TarsosDSP itself has no dependencies, it also did not have any advantages so i reverted back to pure ant. Perhaps I should look into it again.

Regards

Joren

**Comment by ceefour** [comment]
Can you summarize a bit what kind of issues you were having with Maven?

**Comment by ceefour** [comment]
This is one I see: https://github.com/JorenSix/TarsosDSP/commit/ba1614b9ba1c8401140f58fe50e713452c2506db

"uniformity with other Tarsos projects", IMHO uniformity is not a technical issue, so it should be OK. The main gain with Maven here is ease of use from other projects (since using Maven means it makes it easier to publish to Maven Central, note that you can still publish to Maven central with Ant, but it's just more work). Also you can build a Maven project from a "parent" Ant project if that's your worry.

**Comment by JorenSix** [comment]
The main issue I have with it is that I got the feeling I was more and more maintaining a build system than actually doing something meaningful. Since I am not familiar with Maven and the central repository it already took a lot of time to get somewhat familiar with it, and I had the feeling that it would take a lot more time to master it. I actually published something on https://oss.sonatype.org, which to me seems a horrible place (the manual to get something on there was very convoluted). 

Since I rather spend time on functionality and I am the only maintainer of this software package I did not feel it was worth it. So I reverted back to one simple Ant build.xml file and two bash scripts to release the software on my own server. It seems that people are able to find it. Uniformity with other project I'm working on saves me some time.

**Comment by reisners** [comment]
Unfortunately this will limit adoption of TarsosDSP somewhat, I'm afraid, since the rest of world seems to have adopted maven. When I try out a new library, I always look for a maven dependecy first because it's so much easier to integrate in a project. It's a positive feedback loop. If your library is not, that makes it kind of an alien thing that is more hassle to integrate and maintain. I understand that there's a learning curve and some additional work for you, but if you want wide adoption of your product, you should try to save the time of others as well. Just my 2c. And now I'm going to try out TarsosDSP anyway ;-)

**Comment by jobez** [comment]
to the uninitiated, its possible to install the jar into a local maven repository by

- running 'ant' in build/
which should emit the jar in the same directory
- then running
    `mvn install:install-file -Dfile=TarsosDSP-<version>.jar -DgroupId=jorensix -Dversion=<version> -DartifactId=TarsosDSP -Dpackaging=jar`

**Comment by dpoisson2017** [comment]
Thanks axet, this really makes is easy to incorporate into a new project!

**Comment by lukaz-sampaio** [comment]
An alternative to maven is [jitpack.io](https://jitpack.io/). It is more common for Android libraries, but works for any .jar file.
